pandas/types/cast.py

```diff
@@ -886,3 +908,19 @@ def _find_common_type(types):
         return np.dtype('timedelta64[ns]')
 
     return np.find_common_type(types, [])
+
+
+def _cast_scalar_to_array(shape, value, dtype=None):
+    """
+    create np.ndarray of specified shape and dtype, filled with values
+    """
```

Comment:
can you add some tests for this.
I would also raise a helpful error if this is NOT a numpy dtype (and tests for same). 

*though* we *could* handle it (e.g. make it an object array), but that's for another PR (it *might* make some logic simpler), but can do later.